book/src/configuration.md

```diff
@@ -64,6 +64,7 @@ Its settings will be merged with the configuration directory `config.toml` and t
 | `text-width` | Maximum line length. Used for the `:reflow` command and soft-wrapping if `soft-wrap.wrap-at-text-width` is set | `80` |
 | `workspace-lsp-roots` | Directories relative to the workspace root that are treated as LSP roots. Should only be set in `.helix/config.toml` | `[]` |
 | `default-line-ending` | The line ending to use for new documents. Can be `native`, `lf`, `crlf`, `ff`, `cr` or `nel`. `native` uses the platform's native line ending (`crlf` on Windows, otherwise `lf`). | `native` |
+| `insert-final-newline` | Whether to automatically insert a final newline on write if missing | `false` |
```

Comment:
```suggestion
| `insert-final-newline` | Whether to automatically insert a trailing line-ending on write if missing | `false` |
```

We pretty consistently use "line-ending" throughout the codebase and docs. As you say EditorConfig & VSCode are using this option name so I think it's ok to keep the option name as `insert-final-newline` but let's put line-ending in the description. That should also make this a little more discoverable when searching the docs.

(Here and `helix-view/src/editor.rs:290`)